aikido_perception/include/aikido/perception/AprilTagsModule.hpp

```diff
@@ -0,0 +1,68 @@
+/** 
+ * @file AprilTagsModule.hpp
+ * @author Shushman Choudhury
+ * @date Apr 14, 2016
+ * @brief The header for the April Tags detector. The function detectObjects
+ * waits for the next ROS message on the specified topic.
+ */
+
+#ifndef AIKIDO_PERCEPTION_APRILTAGSMODULE_H
+#define AIKIDO_PERCEPTION_APRILTAGSMODULE_H
+
+#include <string>
+#include <ros/ros.h>
+#include <ros/forwards.h>
+#include <ros/single_subscriber_publisher.h>
+#include <aikido/perception/ConfigDataLoader.hpp>
+#include <tf/transform_listener.h>
+#include "yaml-cpp/yaml.h"
+
+#include <dart/dart.h>
+#include <aikido/util/CatkinResourceRetriever.hpp>
+
+#include "PerceptionModule.hpp"
+
+
+namespace aikido{
+namespace perception{
+
+/// The April Tag detector 
+class AprilTagsModule : public virtual PerceptionModule
+{
+public:
+	AprilTagsModule(ros::NodeHandle node, std::string markerTopic, std::shared_ptr<ConfigDataLoader> configData,
+					const dart::common::ResourceRetrieverPtr& resourceRetriever,
+					std::string destinationFrame, dart::dynamics::Frame* referenceLink);
+	virtual ~AprilTagsModule() = default;
+
+
+	void detectObjects(std::vector<dart::dynamics::SkeletonPtr>& skeleton_list,double _timeout=10.0, ros::Time timestamp=ros::Time::now()) override; 
```

Comment:
I'm not sure what to do here then? I've tentatively made it `ros::Time(0.0)`. What did you have in mind instead? To not have a default argument for it?